### PR TITLE
Allow ReadWriteOncePod DVs

### LINKED ISSUE #2365: Support ReadWriteOncePod for DataVolumes
There's a new pvc accessMode called ReadWriteOncePod which means that a PVC can only be accessed by one pod at a time. The ReadWriteOnce accessMode only guaranteed that a single hosted accessed the PVC at a time, but multiple pods could.

It's possible this is as simple as letting the new AccessMode pass validation as outlined in the patch below.

```
diff --git a/pkg/apiserver/webhooks/datavolume-validate.go b/pkg/apiserver/webhooks/datavolume-validate.go
index 690dbb112..4e28fe5e8 100644
--- a/pkg/apiserver/webhooks/datavolume-validate.go
+++ b/pkg/apiserver/webhooks/datavolume-validate.go
@@ -131,10 +131,10 @@ func (wh *dataVolumeValidatingWebhook) validateDataVolumeSpec(request *admission
                // here in storage spec we allow empty access mode and AccessModes with more than one entry
                accessModes := spec.Storage.AccessModes
                for _, mode := range accessModes {
-                       if mode != v1.ReadWriteOnce && mode != v1.ReadOnlyMany && mode != v1.ReadWriteMany {
+                       if mode != v1.ReadWriteOnce && mode != v1.ReadOnlyMany && mode != v1.ReadWriteMany && mode != v1.ReadWriteOncePod {
                                causes = append(causes, metav1.StatusCause{
                                        Type:    metav1.CauseTypeFieldValueInvalid,
-                                       Message: fmt.Sprintf("Unsupported value: \"%s\": supported values: \"ReadOnlyMany\", \"ReadWriteMany\", \"ReadWriteOnce\"", string(accessModes[0])),
+                                       Message: fmt.Sprintf("Unsupported value: \"%s\": supported values: \"ReadOnlyMany\", \"ReadWriteMany\", \"ReadWriteOnce\", \"ReadWriteOncePod\" ", string(accessModes[0])),
                                        Field:   field.Child("PVC", "accessModes").String(),
                                })
                                return causes
```

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2365 

**Special notes for your reviewer**:

**Release note**:

```release-note
ReadWriteOncePod is now allowed to use with CDI DataVolumes
```

### CODE REVIEW

**Comment by akalenyu** [comment]
With the dummy commit flipping lots of tests to use RWOP, we were doing pretty good:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/3093/pull-containerized-data-importer-e2e-ceph/1752087294061318144

There are only 2 failures and those are just tests that use the "local" PVs which only have ReadWriteOnce in the supported access modes. One of the tests could even get away without using "local", so it's dropped in one of the commits.

**Comment by akalenyu** [comment]
/cc @awels @alromeros
ready for reviews

File: tests/datavolume_test.go [hunk 1]
```diff
@@ -1246,30 +1246,12 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
...
-			Entry("[test_id:8045]for clone DataVolume", createCloneDataVolume, fillCommand),
-		)
-
-		DescribeTable("Should pass all DV labels and annotations to the PVC", func(dvFunc func(string, string, string) *cdiv1.DataVolume, url string) {
-			dataVolume := dvFunc("pass-all-labels-dv", "1Gi", url)
+		DescribeTable("should have an alert suppressing label and inherit labels & annotations from DV on corresponding PVC", func(dvFunc func(string, string, string) *cdiv1.DataVolume, url string) {
+			dataVolume := dvFunc("dv-labels-behaviour", "1Gi", url)
 			dataVolume.Labels = map[string]string{"test-label-1": "test-label-1", "test-label-2": "test-label-2"}
 			dataVolume.Annotations = map[string]string{"test-annotation-1": "test-annotation-1", "test-annotation-2": "test-annotation-2"}
+			// Stir things up with this non widely used access mode
```
Comment (alromeros) [inline tests/datavolume_test.go:1253]: Do you plan to add more testing?

Comment (akalenyu) [reply tests/datavolume_test.go:1253]: Nope, as you can see I am even not adding new tests, rather, integrating RWOP into existing ones
I think we shouldn't be too worried about people using RWOP, just allowing it and testing a few select flows

**Review by awels — APPROVED** [review]

**Pull request merged**